src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
             return;
         }
-        Entry<K, V> newEntry;
-        if (data.size() == cap) {
+        Entry<K, V> newEntry = new Entry<>(); 
+        if (data.size() >= cap) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2211218148>#c206b8</a><span> - 07/05/2024, 05:54 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The line 'Entry&lt;K, V&gt; newEntry = new Entry<>();' initializes a new entry, but this entry is immediately replaced if the cache size exceeds the capacity. This results in unnecessary object creation.
</div>
<div id="fix">
<b>Fix:</b> Initialize 'newEntry' only if the cache size is less than the capacity. Otherwise, use the entry returned by the 'evict' method.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -85,7 +85,7 @@
      }
      Entry<K, V> newEntry;
      if (data.size() >= cap) {
          newEntry = evict();
          data.remove(newEntry.getKey());
      } else {
          newEntry = new Entry<>();
      }
```
</div>
</details>
</div>
<br>


</div>